### PR TITLE
Handle multiple geometries in a .dae meshfile_geometry

### PR DESCRIPTION
**Pull request by tkoolen**
On top of #55.

This seems to be pretty slow but it almost works. I'm now getting

image

with https://github.com/tkoolen/universal-robot-dae-jl. Note the missing mesh at the base.

### CODE REVIEW

**Comment by tkoolen** [comment]
I think a likely cause for the missing base link geometry is that I'm skipping `child.type == "Group"`. Also I think we'd ideally want to not even use `mergeBufferGeometries` since it appears to be very slow, instead adding the individual geometries to the scene graph. But they each need a unique uuid; they can't just share `geom.uuid`.

File: src/index.js [hunk 1]
```diff
@@ -61,9 +62,21 @@ function handle_special_geometry(geom) {
         } else if (geom.format == "dae") {
             let loader = new THREE.ColladaLoader();
             let obj = loader.parse(geom.data);
-            let result = obj.scene.children[0].geometry;
-            result.uuid = geom.uuid;
-            return result;
+            let buffer_geometries = []
+            for (let child of obj.scene.children) {
```
Comment (VladimirIvan) [inline src/index.js]: Some collada files have hierarchy deeper than 1 layer. The base link may be one of them.

File: src/index.js [hunk 2]
```diff
@@ -61,9 +62,21 @@ function handle_special_geometry(geom) {
...
+                if (child.type == "Mesh") {
+                    let geometry = child.geometry;
+                    if (geometry.type == "BufferGeometry") {
+                        buffer_geometries.push(geometry);
```
[the remark this answers is not in the document]
Comment (tkoolen) [reply src/index.js]: Ah, you're right. At first I was looking at [`mergeMesh`](https://threejs.org/docs/index.html#api/en/core/Geometry.mergeMesh), which automatically applies the matrix.

File: src/index.js (hunk 1 above)
Comment (VladimirIvan) [reply src/index.js]: I think we may have implemented the same feature in parallel. I was just finishing it up when you created the PR :)

Comment (tkoolen) [reply src/index.js]: If yours is in a better state, please do open it!

Comment (VladimirIvan) [reply src/index.js]: I added texture and material loading as well.
See #58.

**Comment by tkoolen** [comment]
Closing in favor of #58.

**Pull request closed without merging**